bundler/lib/bundler/settings.rb

```diff
@@ -90,14 +90,20 @@ class Settings
     def initialize(root = nil)
       @root            = root
       @local_config    = load_config(local_config_file)
-      @env_config      = ENV.to_h.select {|key, _value| key =~ /\ABUNDLE_.+/ }
+      @env_config      = ENV.to_h
+      @env_config.select! {|key, _value| key.start_with?("BUNDLE_") }
```

Comment:
this is technically different behavior, `BUNDLE_` would not match the regexp but would match the `start_with?`, but I doubt it matters

Reply:
Good catch. If that is the case, we can reject/delete it explicitly here. Or handle it in the other spot you mentioned down in `all`

Reply:
Some ideas:

```ruby
@env_config.select! {|key, _value| key.start_with?("BUNDLE_") && key != "BUNDLE_" }
# or
@env_config.select! {|key, _value| key.start_with?("BUNDLE_")}
@env_config.delete("BUNDLE_")
```

Reply:
Second one look good. Then you're not checking each loop.